lib/bindConnectors/bindMessage.js

```diff
@@ -76,11 +156,45 @@ module.exports = function (message, threadneedle, connectorConfig) {
 
 				threadneedle[methodName](body)
 
-				.done(resolve, reject);
+				.done(function (data) {
+					requestEnd(startTime, { connectorName, operationName: methodName, connectorVersion: connectorConfig.version, status: 'success' }, datadogInstance);
+					resolve(data);
+				}, function (data) {
+					if (process.env.ENABLE_LOGGING === 'true' && logger) {
+						logger.error(
+							data,
+							{
+								connectorName: connectorConfig.name,
+								connectorVersion: connectorConfig.version,
+								operation: event.header.message,
+								requestUrl: `${connectorConfig.globalModel.baseUrl(event.body)}${message.model.url}`,
+								workflowId: event.id,
+							}
+						);
+					}
+					requestEnd(startTime, { connectorName, operationName: methodName, connectorVersion: connectorConfig.version, status: 'failure' }, datadogInstance);
+					reject(data);
+				});
 
 			}
 
 		});
 	};
 
+	// Bind the method to the internal handler object - this is how the connector
+	// processes the inbound messages below.
+	return function (event) {
+		const startTime = performance.now();
+		const dataDogInitialisation = when(initDataDogApi());
```

Comment:
there's no need to wrap this in `when()` - unless you need the functionality from `when`, `initDataDogApi()` is already a promise

Reply:
Without when() the running a connector operation fails with this error:
`TypeError: hooks[event.header.message](...).done is not a function`